### PR TITLE
Manual Webhook Followup: UI Sync [#1326]

### LINKED ISSUE #1326: Manual Webhook Backend Improvements Followup
### Is your feature request related to a specific problem?

This ticket just represents the backend changes that we're making for manual webhooks as a result of testing the front end and backend in combination.

### Describe the solution you'd like

_A description of what you want to happen._

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by pattisdr**
# Purpose

Adjust some backend pieces as we test manual webhooks from a UI perspective. 

# Changes
- Allow Manual Webhooks to be tested - by "testing" we just verify that a valid manual webhook config exists, with fields included.
- Allow DSR Package labels to be supplied as empty strings, but then replace with an underscored/snake cased version of the PII field if this happens (requested by Chris)
- Only run manual webhooks that have valid field configurations

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [ ] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [ ] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [ ] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [x] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #1326

### CODE REVIEW

File: tests/ops/models/test_manual_webhook.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    def test_get_enabled_no_enabled_webhooks(self, db):
+        assert AccessManualWebhook.get_enabled(db) == []
+
+    def test_get_enabled_webhooks(self, db, access_manual_webhook):
+        assert AccessManualWebhook.get_enabled(db) == [access_manual_webhook]
+
+    def test_get_enabled_webhooks_connection_config_disabled(
+        self, db, access_manual_webhook, integration_manual_webhook_config
+    ):
```
Comment (sanders41) [inline tests/ops/models/test_manual_webhook.py:24]: `access_manual_webhook` doesn't get used

```suggestion
    def test_get_enabled_webhooks_connection_config_disabled(
        self, db, integration_manual_webhook_config
    ):
```

File: tests/ops/models/test_manual_webhook.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+class TestManualWebhook:
+    def test_manual_webhook(
+        self, integration_manual_webhook_config, access_manual_webhook
+    ):
+        assert (
+            integration_manual_webhook_config.access_manual_webhook
+            == access_manual_webhook
+        )
+        assert (
+            access_manual_webhook.connection_config == integration_manual_webhook_config
```
[the remark this answers is not in the document]
Comment (pattisdr) [reply tests/ops/models/test_manual_webhook.py:13]: I was just verifying the relationship was set up properly so it wasn't a list!

File: src/fidesops/ops/schemas/manual_webhook_schemas.py [hunk 3]
```diff
@@ -9,18 +9,24 @@
...
 
-class ManualWebhookFieldType(ConstrainedStr):
+class PIIFieldType(ConstrainedStr):
     """Using ConstrainedStr instead of constr to keep mypy happy"""
 
     min_length = 1
     max_length = 200
 
 
+class DSRLabelFieldType(ConstrainedStr):
```
[the remark this answers is not in the document]
Comment (pattisdr) [reply src/fidesops/ops/schemas/manual_webhook_schemas.py:19]: Yes, Chris asked for me to change this to let that field accept an empty string, but I can't let them both be empty strings.

File: tests/ops/models/test_manual_webhook.py (hunk 1 above)
Comment (pattisdr) [reply tests/ops/models/test_manual_webhook.py:24]: it does get used, it needs to exist here! I need to remember the fixture you showed me before to denote this.

Comment (pattisdr) [reply tests/ops/models/test_manual_webhook.py:24]: ah yes, @pytest.mark.usefixtures("access_manual_webhook")

File: tests/ops/api/v1/endpoints/test_manual_webhooks.py [hunk 4]
```diff
@@ -420,3 +439,74 @@ def test_get_manual_webhooks(
...
+        url,
+        generate_auth_header,
+        integration_manual_webhook_config,
+    ):
+        auth_header = generate_auth_header([CONNECTION_READ])
+
+        response = api_client.get(url, headers=auth_header)
+        assert 200 == response.status_code
+        assert response.json()["test_status"] == "failed"
+        assert response.status_code == 200
```
[the remark this answers is not in the document]
Comment (pattisdr) [reply tests/ops/api/v1/endpoints/test_manual_webhooks.py:478]: whoops

File: src/fidesops/ops/schemas/manual_webhook_schemas.py [hunk 5]
```diff
@@ -9,18 +9,24 @@
...
+    max_length = 200
+
+
 class ManualWebhookField(BaseSchema):
     """Schema to describe the attributes on a manual webhook field"""
 
-    pii_field: ManualWebhookFieldType
-    dsr_package_label: Optional[ManualWebhookFieldType] = None
+    pii_field: PIIFieldType
+    dsr_package_label: Optional[DSRLabelFieldType] = None
```
[the remark this answers is not in the document]
Comment (pattisdr) [reply src/fidesops/ops/schemas/manual_webhook_schemas.py:29]: sure, added. empty strings are converted to a lower cased/snake cased version of the pii_field in the root validator if they don't exist, but doesn't hurt to add.

**Review by sanders41 — APPROVED** [review]

**Comment by pattisdr** [comment]
thank you @sanders41!

**Pull request merged**